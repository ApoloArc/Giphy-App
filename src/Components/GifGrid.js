import { GifTemplate } from './GifTemplate';
import { useFetchGifs } from './Hooks/useFetchsGifs';
import PropTypes from 'prop-types';


const GifGrid = ({category}) => {
    
    const {data:images, loading} = useFetchGifs(category);

    return (
        <div>
            <h3 className='titulo-category'>{category}</h3>
            {loading && <p className='animate__animated animate__fadeInDown'>Cargando, espera compare...</p>}
            <div className='grid'>
            {
               images.map(img => <GifTemplate key={img.id} {...img}/>)
            }
            </div>
        </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
