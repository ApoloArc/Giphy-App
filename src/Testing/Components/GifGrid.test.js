import {shallow} from 'enzyme';
import GifGrid from '../../Components/GifGrid';
import { useFetchGifs } from '../../Components/Hooks/useFetchsGifs';
import '@testing-library/jest-dom'


describe('Pruebas para el GifGid', () => {
    
    const category = 'Hola Mundo';
    const wrapper = shallow(<GifGrid category={category}/>)

    test('debe de tomar el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})