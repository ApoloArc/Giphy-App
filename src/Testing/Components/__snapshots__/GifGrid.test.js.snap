// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el GifGid debe de tomar el snapshot 1`] = `
<div>
  <h3
    className="titulo-category"
  >
    Hola Mundo
  </h3>
  <p
    className="animate__animated animate__fadeInDown"
  >
    Cargando, espera compare...
  </p>
  <div
    className="grid"
  />
</div>
`;
